### PR TITLE
feat: Integrate Tanzu Sources (vSphere/Horizon) for Knative

### PR DESCRIPTION
**Pull request by lamw**
## Summary

This change refactors VEBA to now use Tanzu Sources for Knative when configuring either vSphere or Horizon sources and maintains event-router for Webhook processor configurations. 

In addition, there were few optimizations that were needed including updating the VEBA VM hardware configuration which had some intermittent build errors preventing consistent builds.

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

Closes: #1066

## Testing Verification

* Successfully deployed VEBA appliance using various configurations including: vSphere, vSphere + Webhook, vSphere + Horizion + Webhook Processor

## Additional Information

* Any other details you wish to include or mention

If you have any questions/comments, feel free to reach out to team on Slack [#vcenter-event-broker-appliance](https://vmwarecode.slack.com/archives/CQLT9B5AA)

Thank you from the VEBA Team! 🥳

### CODE REVIEW

File: files/configs/horizon-source/templates/horizon-source-template.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  name: horizon-source
+spec:
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1
+      kind: Broker
+      name: default
+      namespace: vmware-functions
+  address: #@ "https://"+horizon
+  #@ if horizonTls == "True":
```
Comment (embano1) [inline files/configs/horizon-source/templates/horizon-source-template.yml:20]: Just to double-check: `if horizonTls == "True"` means disable TLS or enable TLS?

File: files/setup-06-vsphere-sources.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# SPDX-License-Identifier: BSD-2
+
+# Setup vSphere Sources
+
+set -euo pipefail
+
+echo -e "\e[92mCreating vSphere Secret ..." > /dev/console
+# Create vSphere Secret
+if [ ${VCENTER_DISABLE_TLS} == "True" ]; then
+    kn vsphere auth create \
```
Comment (embano1) [inline files/setup-06-vsphere-sources.sh:12]: Bit weird that you deploy most of it using YAML/ytt and here use a CLI. Why not use manifests?

File: files/configs/horizon-source/templates/horizon-source-template.yml (hunk 1 above)
Comment (lamw) [reply files/configs/horizon-source/templates/horizon-source-template.yml:20]: That's correct, I most likely kept the variable name "simple" when working on vSphere sources from event-router, so just dup'ed variable name here but it is indeed reading from https://github.com/vmware-samples/vcenter-event-broker-appliance/blob/development/files/setup.sh#L34 

Happy to modify for clarity purposes

File: files/setup-08-tinywww.sh [hunk 3]
```diff
@@ -13,6 +13,11 @@ VEBA_CONFIG_FILE=/root/config/veba-config.json
 TINYWWW_TEMPLATE=/root/config/tinywww/templates/tinywww-template.yaml
 TINYWWW_CONFIG=/root/config/tinywww/tinywww.yaml
 
+# Basic Auth for TinyWWW endpoints
+kubectl -n vmware-system create secret generic basic-auth \
```
Comment (embano1) [inline files/setup-08-tinywww.sh:17]: remind me, why and where is this needed now?

File: files/setup-06-vsphere-sources.sh (hunk 2 above)
Comment (lamw) [reply files/setup-06-vsphere-sources.sh:12]: I started out w/vSphere Sources which provided kn plugin ... I guess looking at Horizon Sources, I had no choice but perhaps might be good to just keep it consistent. I wasn't sure if there was a preference to leverage the plugin over manifests

File: files/setup-05-knative.sh [hunk 4]
```diff
@@ -55,4 +57,23 @@ SOCKEYE_CONFIG=/root/config/knative/sockeye.yaml
...
+
+# Wait for all controllers/webhook to be ready
+kubectl wait --for=condition=ready pod -l app=horizon-source-controller --timeout=${KUBECTL_WAIT} -n vmware-sources
+kubectl wait --for=condition=ready pod -l app=horizon-source-webhook --timeout=${KUBECTL_WAIT} -n vmware-sources
+kubectl wait --for=condition=ready pod -l app=vsphere-source-webhook --timeout=${KUBECTL_WAIT} -n vmware-sources
+
+# Enable debug logs for tanzu sources
+if [ ${TANZU_SOURCES_DEBUG} -eq "True" ]; then
+    kubectl -n vmware-sources get cm config-logging -o yaml > /tmp/vmware-sources-config-logging.yaml
+    sed -i 's/"level": "info"/"level": "debug"/g' /tmp/vmware-sources-config-logging.yaml
```
Comment (embano1) [inline files/setup-05-knative.sh:78]: Is this used/applied anyhwere? Not sure I'm understanding what this is supposed to do.

File: files/setup-08-tinywww.sh (hunk 3 above)
Comment (lamw) [reply files/setup-08-tinywww.sh:17]: To access any of the endpoints served by TinyWWW - /events, /bootstrap, etc.

Basically, we don't want someone to be able to view events or debug logs w/o some sort of auth

File: files/setup-05-knative.sh (hunk 4 above)
Comment (lamw) [reply files/setup-05-knative.sh:78]: 🦅 👁️ No, this was temp way of enabling debugging w/sources. Thanks to Gaberiel, this is now part of YTT setup for Sources

Let me remove

File: files/configs/horizon-source/templates/horizon-source-template.yml (hunk 1 above)
Comment (embano1) [reply files/configs/horizon-source/templates/horizon-source-template.yml:20]: Yeah, would be great to have those two align, i.e. if `horizonTls == "True"` `skipTLSVerify` should be `false`

File: files/setup-06-vsphere-sources.sh (hunk 2 above)
Comment (embano1) [reply files/setup-06-vsphere-sources.sh:12]: Yeah, no benefit I guess in using the CLI, so let's go for consistency.

File: files/setup-08-tinywww.sh (hunk 3 above)
Comment (embano1) [reply files/setup-08-tinywww.sh:17]: ACK. And this is not handled by an Ingress? I thought we had those endpoints exposed through ingress which also has auth?

File: files/setup-05-knative.sh (hunk 4 above)
Comment (embano1) [reply files/setup-05-knative.sh:78]: Haha, ok. 🦅 👁️ can still do it 🤣

Comment (lamw) [reply files/setup-05-knative.sh]: Resolved

File: files/setup-08-tinywww.sh (hunk 3 above)
Comment (lamw) [reply files/setup-08-tinywww.sh:17]: No, we've always had it like this

File: files/setup-06-vsphere-sources.sh (hunk 2 above)
Comment (lamw) [reply files/setup-06-vsphere-sources.sh]: Resolved

File: files/configs/horizon-source/templates/horizon-source-template.yml (hunk 1 above)
Comment (lamw) [reply files/configs/horizon-source/templates/horizon-source-template.yml]: Resolved

**Review by embano1 — APPROVED** [review]
LGTM

**Pull request merged**